lib/tasks/assets.rake

```diff
@@ -1,28 +1,38 @@
-namespace :deploy do
-  # Downloads a tar from a URL, extracts its contents into ./tmp,
-  # and returns a colon separated list of paths belonging to the first level directories
-  # inside the original tar file.
+namespace :assets do
+
+  task :download_and_precompile, [:url_template, :credentials, :branch] => :environment do |t, args|
+    if args[:url_template].blank?
+      puts "Not including any external assets"
+      next
+    end
+    target_path = "./tmp/assets_source"
+    FileUtils.mkdir_p target_path
+    Rake::Task['assets:download_external_assets'].invoke(target_path, args[:url_template], args[:credentials], args[:branch])
+    Rake::Task['assets:precompile_assets'].invoke(target_path)
```

Comment:
if all the precompile_assets task does anyway is call a different rake task, why not just invoke it here?